### ISSUE TITLE
Datepicker value not clearing when hitting backspace

### ISSUE THREAD

**Issue by clye-cog** [GovTechSG/sgds-govtech-react#267]
### Prerequisites

- [X] I have [searched](https://github.com/GovTechSG/sgds-govtech-react/issues) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/GovTechSG/sgds-govtech-react/blob/v2/CONTRIBUTING.md)

### Describe the issue

- Basically, it will not actually clear the date value if we hit backspace even though it is visually cleared.
- However, the field correctly clears when clicking "x".
- See the video:

https://github.com/user-attachments/assets/1a6ac320-8773-4244-8385-6b1ca9dcfbfa

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### Describe your frontend stack. What version of React and @govtechsg/sgds-react are you using? CSR or SSR?

React Typescript NextJS

**Comment by clukhei** [comment]
@clye-cog could you also share your code on how are you accessing the value of the datepicker?

**Comment by clye-cog** [comment]
Hi @clukhei , we are currently using it like this:

```ts
              <SgdsDatePicker
                id="form-date-picker"
                initialValue={initialValue}
                displayDate={displayDate}
                onChangeDate={field.onChange}
                className={error ? "is-invalid" : ""}
                invalidFeedback={
                  overrideDefaultInvalidMsg ? "" : "Please enter a valid date"
                }
                required={false}
              />
```

We use react-hook-form to control this component

**Comment by clukhei** [comment]
@clye-cog The fixed is released in v2.7.2.

i will close this issue now. Feel free to reopen again if you still face issues.
